### PR TITLE
TSPS-316 remove all unhelpful "any()" matchers from our tests

### PR DESCRIPTION
**Pull request by jsotobroad**
### Description 

I got rid of pretty much all of them and I probably went overboard with some tests because we dont need to test all arguments match EVERY single time but it was easier to copy paste once i was in there.  Especially for the methods that are testing for exceptions being thrown you can use `any()` cuz you don't really care.  Basically anytime you do a positive test it should try to not use `any()` as much as possible

### Jira Ticket
https://broadworkbench.atlassian.net/browse/TSPS-316

### CODE REVIEW

File: service/src/test/java/bio/terra/pipelines/dependencies/sam/SamServiceTest.java [hunk 1]
```diff
@@ -103,33 +102,26 @@ void getServiceAccountToken() throws IOException {
...
-    when(adminApi.adminGetUserByEmail(any())).thenReturn(new UserStatus());
-    when(samClient.adminApi(any())).thenReturn(adminApi);
-
-    SamService samService = spy(new SamService(samClient));
+    when(adminApi.adminGetUserByEmail("blahEmail")).thenReturn(new UserStatus());
+    when(samClient.adminApi("blahToken")).thenReturn(adminApi);
 
-    samService.checkAdminAuthz(new SamUser("blah", "blah", new BearerToken("blah")));
+    samService.checkAdminAuthz(
+        new SamUser("blahEmail", "blahSubjectId", new BearerToken("blahToken")));
```
Comment (mmorgantaylor) [inline service/src/test/java/bio/terra/pipelines/dependencies/sam/SamServiceTest.java:116]: super nit but can we change "blahSubjectId" to "doesnt matter" ?

File: service/src/test/java/bio/terra/pipelines/stairway/imputation/steps/azure/SubmitCromwellRunSetStepTest.java [hunk 2]
```diff
@@ -137,14 +141,14 @@ void doStepNoMatchingMethod() {
...
-    when(cbasService.getAllMethods(any(), any())).thenReturn(getAllMethodsResponse);
+    when(cbasService.getAllMethods("cbasUri", "thisToken")).thenReturn(getAllMethodsResponse);
 
     // do the step
     SubmitCromwellRunSetStep submitCromwellRunSetStep =
         new SubmitCromwellRunSetStep(cbasService, samService, pipelinesService, cbasConfiguration);
     StepResult result = submitCromwellRunSetStep.doStep(flightContext);
 
-    // make sure the step was a fatal faiilure
+    // make sure the step was a fatal failure
```
Comment (mmorgantaylor) [inline service/src/test/java/bio/terra/pipelines/stairway/imputation/steps/azure/SubmitCromwellRunSetStepTest.java:151]: <3

**Review by mmorgantaylor — APPROVED** [review]
this is SO much better. a couple minor questions but 👍

File: service/src/test/java/bio/terra/pipelines/dependencies/gcs/GcsServiceTest.java [hunk 3]
```diff
@@ -96,16 +109,15 @@ void socketExceptionRetriesEventuallySucceed() {
 
     when(mockStorageService.signUrl(
             any(BlobInfo.class),
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/test/java/bio/terra/pipelines/dependencies/gcs/GcsServiceTest.java:111]: yea exactly, this is why i was saying i went a little overboard where EVERYWHERE i put all the specific checkers/captures but really it should just be in the happy path tests.  In these tests it should be necessary if you have decent code coverage.  LIke a happy, sad, and a little in between tests

File: service/src/test/java/bio/terra/pipelines/dependencies/leonardo/LeonardoServiceTest.java [hunk 4]
```diff
@@ -48,13 +52,12 @@ void init() {
...
     List<ListAppResponse> expectedResponse = List.of(new ListAppResponse());
 
-    LeonardoClient leonardoClient = mock(LeonardoClient.class);
     AppsApi appsApi = mock(AppsApi.class);
     when(appsApi.listAppsV2(workspaceId, null, null, null))
         .thenAnswer(errorAnswer)
         .thenReturn(expectedResponse);
 
-    LeonardoService leonardoService = spy(new LeonardoService(leonardoClient, appUtils, template));
+    leonardoService = spy(new LeonardoService(leonardoClient, appUtils, template));
```
[the remark this answers is not in the document]
Comment (jsotobroad) [reply service/src/test/java/bio/terra/pipelines/dependencies/leonardo/LeonardoServiceTest.java:60]: ooo i like this jsut putting it in the init file.  and exactly we use spy cuz its way easier to mock the leonardo call than anything else.  I would probably recommend pushing it down into the client so its mockable liek the rest of our services if we still using this service regularly

[the remark this answers is not in the document]
Comment (mmorgantaylor) [reply service/src/test/java/bio/terra/pipelines/dependencies/leonardo/LeonardoServiceTest.java:60]: makes sense, sounds good!

File: service/src/test/java/bio/terra/pipelines/dependencies/gcs/GcsServiceTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmorgantaylor) [reply service/src/test/java/bio/terra/pipelines/dependencies/gcs/GcsServiceTest.java:111]: that makes sense, thanks!

**Pull request merged**